runners/direct-java/src/test/java/org/apache/beam/runners/direct/KeyedPValueTrackingVisitorTest.java

```diff
@@ -117,26 +80,17 @@ public void noInputUnkeyedOutput() {
   }
 
   @Test
-  public void keyedInputNotProducesKeyedOutputUnkeyedOutput() {
-    PCollection<Integer> onceKeyed =
-        p.apply(Create.<Integer>of(1, 2, 3))
-            .apply(new PrimitiveKeyer<Integer>())
-            .apply(ParDo.of(new IdentityFn<Integer>()));
+  public void keyedInputWithoutKeyPreserving() {
```

Comment:
Since there are no key-preserving transforms I can't test it without adding abstractions that we ultimately won't need. Rebased on top of this, I do have a PR that alters stateful `ParDo` to have a key-preserving intermediate function, but I was throwing out this refactor first since it was trivial. I could add all that to this one if you like.